packages/@aws-cdk/aws-ecr-assets/lib/tarball-asset.ts

```diff
@@ -78,7 +78,7 @@ export class TarballImageAsset extends CoreConstruct implements IAsset {
       executable: [
         'sh',
         '-c',
-        `docker load -i ${relativePathInOutDir} | sed "s/Loaded image: //g"`,
+        `docker load -i ${relativePathInOutDir} | sed -nr 's/^Loaded image: (.*)$/\\1/p'`,
```

Comment:
This code will not work on MacOS, as `-r` is not a valid flag for `sed` on that platform.

Reply:
Well that's a bit of a piss off...hmmmm. I'll work on figuring our an alternative. Technically what's really needed is to strip out any following lines. It isn't my ideal, but we could pipe this into `head` which should work better cross platform...

Reply:
@rix0rrr For a test like that to work, I would need to communicate with the docker daemon directly in order to replicate the issue being fixed here. Is there an existing precedent for running shell commands in integ tests? I had pushed back earlier when @madeline-k suggested this as I don't think that this PR is the right place to backfill testing for a completely untested class.